viam-cartographer.go

```diff
@@ -273,32 +274,43 @@ func (cartoSvc *cartographerService) StartDataProcess(
 				return
 			}
 
+			// If we're not using live data, we read from the sensor as fast as
+			// possible, since the sensor is just playing back pre-captured data.
+			if !cartoSvc.useLiveData {
+				cartoSvc.getNextDataPoint(cancelCtx, lidar, c)
+				continue
+			}
+
 			select {
 			case <-cancelCtx.Done():
 				return
 			case <-ticker.C:
-				cartoSvc.activeBackgroundWorkers.Add(1)
 				if err := cancelCtx.Err(); err != nil {
 					if !errors.Is(err, context.Canceled) {
 						cartoSvc.logger.Errorw("unexpected error in SLAM data process", "error", err)
 					}
-					cartoSvc.activeBackgroundWorkers.Done()
 					return
 				}
+
+				cartoSvc.activeBackgroundWorkers.Add(1)
 				goutils.PanicCapturingGo(func() {
 					defer cartoSvc.activeBackgroundWorkers.Done()
-					if _, err := dim2d.GetAndSaveData(cancelCtx, cartoSvc.dataDirectory, lidar, cartoSvc.logger); err != nil {
-						cartoSvc.logger.Warn(err)
-					}
-					if c != nil {
-						c <- 1
-					}
+					cartoSvc.getNextDataPoint(cancelCtx, lidar, c)
 				})
 			}
 		}
 	})
 }
 
+func (cartoSvc *cartographerService) getNextDataPoint(ctx context.Context, lidar lidar.Lidar, c chan int) {
+	if _, err := dim2d.GetAndSaveData(ctx, cartoSvc.dataDirectory, lidar, cartoSvc.logger); err != nil {
```

Comment:
Ok cool, so this will be handled as part of full modularization.

Reply:
Ah yeah I realize I didn't answer the question - afaik it will be handled either by Full Mod _or_ Cloud Story. cc @EshaMaharishi

Reply:
@tessavitabile yes, most likely it will be handled as part of cloud story, since that's where we'll bring back offline mode and start using the replay camera in earnest

Reply:
Thank you!